src/Event/Subscriber/OmnipayGatewayRequestSubscriber.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * payment-gateway-logger
+ *
+ * @package    payment-gateway-logger
+ * @author     manzoj
+ * @version    1
+ */
+
+namespace PaymentGatewayLogger\Event\Subscriber;
+
+use Guzzle\Common\Event;
+use PaymentGatewayLogger\Event\Constants;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
```

Comment:
you're only using Psr\Log to get the log levels and the `log` declaration, right? what do you think about just defining that yourself? Omnipay typically tries to avoid unnecessary packages because they can create version clashes with people's existing codebases.

Reply:
yeah that's correct, I'm only using `log` in my subscriber. The reason why I'm using `Psr\Log` instead of creating my own interface is in case a user decides that they want to set up their own Subscriber class that calls `warn`, `debug` and other such methods. So I think we should keep it as in, wdyt?

Reply:
They could create their own Subscriber if they wanted to do that, right?

Reply:
If you want to use psr/log, can you use ^1.0 instead of ^1.1 to minimize version conflicts?

Reply:
Actually, they'd have to make their own Subscriber class if they wanted to use other methods, wouldn't they?

Omnipay's dependence on an outdated version of Guzzle has caused lots of version conflict issues for us, so I'd definitely recommend avoiding if possible

Reply:
> They could create their own Subscriber if they wanted to do that, right?

Ah yes, good point! In that case I will go with defining my own logging interface to avoid version clashes.

Reply:
Hmm just some thoughts..the guzzle version I'm using already defines `psr\log` 1.0 as a dependency. I'm able to use `psr\log` in this way without explicitly making it a a dependency of this project. In this way I can keep both `use Psr\Log\LoggerInterface` `use Psr\Log\LogLevel`. I'm not an expert on composer so I'm not sure if this is considered a bad practice (relying on your dependicy's dependencies)?

Reply:
okey dokey